src/network/mcpe/handler/LoginPacketHandler.php

```diff
@@ -70,19 +72,36 @@ public function handleLogin(LoginPacket $packet) : bool{
 
 		$clientData = $this->parseClientData($packet->clientDataJwt);
 
-		try{
-			$skin = $this->session->getTypeConverter()->getSkinAdapter()->fromSkinData(ClientDataToSkinDataHelper::fromClientData($clientData));
-		}catch(\InvalidArgumentException | InvalidSkinException $e){
-			$this->session->disconnectWithError(
-				reason: "Invalid skin: " . $e->getMessage(),
-				disconnectScreenMessage: KnownTranslationFactory::disconnectionScreen_invalidSkin()
-			);
+		$this->server->getAsyncPool()->submitTask(new ProcessSkinTask(
+			ClientDataToSkinDataHelper::fromClientData($clientData),
```

Comment:
This doesn’t cover the case when base64 data is invalid. See https://github.com/pmmp/BedrockProtocol/blob/f5daac3c11b7a1f5ba2437ecc09f9ad5ebf72aa8/src/types/login/ClientDataToSkinDataHelper.php#L33